sssd_test_framework/utils/realmd.py

```diff
@@ -0,0 +1,103 @@
+"""Manage realm Operations."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from pytest_mh import MultihostHost, MultihostUtility
+from pytest_mh.conn import ProcessResult
+
+__all__ = [
+    "RealmdUtils",
+]
+
+
+class RealmdUtils(MultihostUtility[MultihostHost]):
+    """
+    Interface to the realm utility.
+
+    .. code-block:: python
+        :caption: Example usage
+
+        @pytest.mark.topology(KnownTopology.AD)
+        def test_realm_discover(client: Client, provider: ADProvider):
+            client.realmd.discover(provider.host.domain, "--use-ldaps"])
+    """
+
+    def discover(self, *, args: list[Any] | None = None) -> ProcessResult:
+        """
+        Discover a realm and it's capabilities.
+
+        :param args: Additional arguments, defaults to None
+        :type args: list[Any] | None, optional
+        """
+        if args is None:
+            args = []
+
+        return self.host.conn.exec(["realm", "discover", *args])
+
+    def leave(self, args: list[Any] | None = None) -> ProcessResult:
+        """
+        Deconfigure or remove a client system from a realm.
```

Comment:
The term realm doesn't need extra description. I still vote for "Discover Realms." I want fewer words, but it's pretty inconsequential. Either way it's fine.

Reply:
This comment looks out of place.